src/main/java/org/broadinstitute/hellbender/tools/spark/sv/InsertSizeMetricsCollectorSpark.java

```diff
@@ -1,99 +1,256 @@
+     * @return           the requested group's stats information, organized by pair orientations
+     * @throws           IllegalArgumentException
+     */
+    public Map<SamPairUtil.PairOrientation, Tuple2<Histogram<Integer>, InsertSizeMetrics>> getHistogramsAndMetrics(final String groupName) throws IllegalArgumentException {
 
-        trimHistogramAndCollectMeanAndSD(lengthRDD, HistogramMADTolerance);
+        final Integer i = mapFromGroupNameToIndexInList.get(groupName);
+        if(null==i){
+            throw new IllegalArgumentException("Requested group name doesn't exist in current data.");
+        }
+        final SortedMap<GroupInfo, Map<SamPairUtil.PairOrientation, Tuple2<Histogram<Integer>, InsertSizeMetrics>>> groupInfo = histogramsAndMetrics.get(i);
+        final Map<SamPairUtil.PairOrientation, Tuple2<Histogram<Integer>, InsertSizeMetrics>> groupStats = groupInfo.values().iterator().next();
+        return groupStats;
     }
 
-    public final InsertSizeMetrics getMetrics(){
-        return metrics;
+    /**
+     * Utility getter for retrieving InsertSizeMetrics of a particular group, given its name.
+     * Stats info returned are organized by pair orientations.
+     * If a particular pair orientation is unavailable (i.e. no reads of this group has pairs of that orientation), it is not returned.
+     * @param groupName String representation of the group's name/id, whose InsertSizeMetrics is requested.
+     * @return          the requested group's InsertSizeMetrics information, organized by pair orientations
+     * @throws IllegalArgumentException
+     */
+    public Map<SamPairUtil.PairOrientation, InsertSizeMetrics> getMetrics(final String groupName) throws IllegalArgumentException {
+        final Map<SamPairUtil.PairOrientation, Tuple2<Histogram<Integer>, InsertSizeMetrics>> intermediate = getHistogramsAndMetrics(groupName);
+        Map<SamPairUtil.PairOrientation, InsertSizeMetrics> result = new HashMap<>();
+        for(Map.Entry<SamPairUtil.PairOrientation, Tuple2<Histogram<Integer>, InsertSizeMetrics>> entry : intermediate.entrySet()){
+            result.put(entry.getKey(), entry.getValue()._2());
+        }
+        return result;
     }
 
-    public final Histogram<Integer> getHitogram(){
-        return htslibHistogram;
+    /**
+     * Utility getter for retrieving InsertSizeMetrics of a particular group, given its name and requested orientation of the pairs.
+     * If no read pairs are available in this particular group of the requested orientation, returns null.
+     * @param groupName   String representation of the group's name/id, whose InsertSizeMetrics information is requested.
+     * @param orientation Requested orientation.
+     * @return            InsertSizeMetrics of the requested group, of the requested orientation (could be null if no reads available)
+     */
+    public InsertSizeMetrics getMetricsOfOrientation(final String groupName, final SamPairUtil.PairOrientation orientation){
```

Comment:
Name should be "getMetricsByGroupNameAndOrientation".

This currently isn't used, do you think it is necessary?